### PR TITLE
Post approve build direct link

### PR DESCRIPTION
**Pull request by shuyama1**
Image's already built and pushed.

Test PR: https://github.com/GoogleCloudPlatform/magic-modules/pull/7672

`Approve Build` status shown up and linked directly to the build needs approval

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .ci/containers/membership-checker/main.go
Comment (shuyama1) [inline .ci/containers/membership-checker/main.go:99]: This change is to prevent labels not being removed for PRs whose contributor joins the org after creating the PR (e.g. https://github.com/GoogleCloudPlatform/magic-modules/pull/7610)

File: .ci/containers/membership-checker/main.go
Comment (shuyama1) [inline .ci/containers/membership-checker/main.go:100]: This change is to prevent labels not being removed for PRs whose contributor joins the org after creating the PR (e.g. https://github.com/GoogleCloudPlatform/magic-modules/pull/7610

Comment (ScottSuarez) [reply .ci/containers/membership-checker/main.go:100]: Does `removeAwaitingApprovalLabel` remove the label only if present?

I worry if it tries to remove something that doesn't exist the call to github will error out.. Although looking into the call it seems just try to remove regardless of presence and the api tolerates it. Is this so?

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: .ci/containers/membership-checker/main.go
Comment (shuyama1) [reply .ci/containers/membership-checker/main.go:100]: It will return a 404 status code for non-exist labels, but won't error out. So I think it should be fine.

File: .ci/containers/membership-checker/community.go
Comment (shuyama1) [reply .ci/containers/membership-checker/community.go:39]: It seems that we can omit global here. I was just trying to align with the url that we use to post other status, but I can totally remove it. What's your thought?

File: .ci/containers/membership-checker/community.go
Comment (shuyama1) [reply .ci/containers/membership-checker/community.go:43]: I was also hesitating about it a bit. Yeah, if we set this to be pending, we'll have to flip it. We can only have the flip step for community PRs, as auto-run PRs don't have this status. But for users who join the membership after they create a PR, this check will always remain pending as there's no way we can change it I assume (my understanding is that if we don't remove the status, it will not be updated and will always show on GH UI).
Plus, I feel like we can consider it more like a button instead of a status, and we still use the state of `changelog-checker` as a indicator.

File: .ci/containers/membership-checker/community.go
Comment (ScottSuarez) [reply .ci/containers/membership-checker/community.go:39]: ack.. Lets keep it!

File: .ci/containers/membership-checker/community.go
Comment (ScottSuarez) [reply .ci/containers/membership-checker/community.go:43]: If on auto run we just post the approval in the generate diffs run irregardless of the presence of pending it should be fine no?

Comment (shuyama1) [reply .ci/containers/membership-checker/community.go:43]: Then, basically we're gonna have `Approve Build` status posted for every PR, which I'm not sure if it's confusing. Plus, if we add the part that posts the approval in the generate diffs, then we need to either add an extra step (slow down the generate diffs) or add the feature in one of the current steps,`github-differ` likely (add more time to generate diffs + rewrite some of the current code to bash). Also we're gonna have the source code of this feature in multiple places and I'm not sure if that's ideal.

I guess we can rename this status to something like `community-checker link` or similar to indicate that the status is not supposed to tell if the build is waiting for approval but just a temporary workaround to post build link before cloud build post it directly. Success state of this status just means the build is posted successfully.

**Review by ScottSuarez — APPROVED** [review]

File: .ci/containers/membership-checker/community.go
Comment (shuyama1) [reply .ci/containers/membership-checker/community.go:43]: As discussed offline, I'll play around with the approaches for implementing pending/approval for the status and will possibly add it in another PR

**Pull request merged**
